frontend/src/components/managePermissions/ManagePermissions.vue

```diff
@@ -46,24 +50,32 @@ const userRoleAssignments = shallowRef<FamApplicationUserRoleAssignmentGet[]>(
     props.userRoleAssignments
 );
 
-const applicationAdmins = shallowRef<
-    FamAppAdminGetResponse[]
->(props.applicationAdmins);
+const applicationAdmins = shallowRef<FamAppAdminGetResponse[]>(
+    props.applicationAdmins
+);
+
+const manageApplicationSelected = shallowRef<FamApplication>(
+    selectedApplication.value as FamApplication
+);
 
 onUnmounted(() => {
     resetNotification();
 });
 
-const onApplicationSelected = async (e: DropdownChangeEvent) => {
-    setSelectedApplication(e.value ? JSON.stringify(e.value) : null);
-    if (e.value.application_id === FAM_APPLICATION_ID) {
+watch(manageApplicationSelected, async (newManageApplicationSelected) => {
```

Comment:
I wonder maybe let's not refactor this application dropdown to a reusable component? cause I feel it complicated the thing a little bit... the one we use here will change a global state variable, the other place we use for grant app admin will change the local form variable. I think maybe it's more clear to just keep them separated.  Cause we use the primevue dropdown anyway, and just pass different data. And if we want to apply the same style, we can give the same class name, and move the style to layout.scss maybe or somewhere that they can share.  Not sure how you feel? thanks!!

Reply:
It seemed a good idea at first to make it a reusable component but I think I agree with you that the complexity has increased a bit. I'll revert it back to what it was and change the new page accordingly. Thanks.